experimental/CliffordAlgebras/src/CliffordAlgebras.jl

```diff
@@ -0,0 +1,584 @@
+canonical basis of its parent Clifford algebra,
+but all coefficients corresponding to basis elements
+with even grading are set to zero. This also updates
+the field `x.odd_coeffs`.
+"""
+function odd_coefficients(x::CliffordAlgebraElem)
+  if isdefined(x, :odd_coeffs)
+    return x.odd_coeffs
+  end
+  _set_even_odd_coefficients!(x)
+  return x.odd_coeffs
+end
+
+################################################################################
+#
+#  String I/O
+#
+################################################################################
+
+### Algebra ###
+function Base.show(io::IO, ::MIME"text/plain", C::CliffordAlgebra)
+  io = pretty(io)
+  print(io, "Clifford algebra of quadratic space with Gram matrix\n")
+  print(io, Indent())
+  show(io, "text/plain", gram_matrix(C))
+  print(io, Dedent(), "\ndefined over $(base_ring(C))")
+end
+
+Base.show(io::IO, C::CliffordAlgebra) = print(io, "Clifford algebra over $(base_ring(C))")
+
+### Elements ###
+function Base.show(io::IO, x::CliffordAlgebraElem)
+  print(io, "[")
+  foreach(y -> print(io, "$y "), coefficients(x)[1:(end - 1)])
+  print(io, "$(coefficients(x)[end])]")
+end
+
+################################################################################
+#
+#  basic functionality
+#
+################################################################################
+
+@doc raw"""
+    zero(C::CliffordAlgebra) -> CliffordAlgebraElem
+
+Return the additive identity of the Clifford algebra $C$.
+"""
+zero(C::CliffordAlgebra) = C()
+
+@doc raw"""
+    one(C::CliffordAlgebra) -> CliffordAlgebraElem
+
+Return the multiplicative identity of the Clifford algebra $C$.
+"""
+function one(C::CliffordAlgebra)
+  v = fill(base_ring(C)(), dim(C))
+  v[1] = base_ring(C)(1)
+  return C(v)
+end
+
+@doc raw"""
+    basis(C::CliffordAlgebra, i::Int) -> CliffordAlgebraElem
+
+Return the $i$-th canonical basis vector of the Clifford algebra $C$.
+"""
+function basis(C::CliffordAlgebra, i::Int)
+  res = CliffordAlgebraElem(C)
+  res.coeffs[i] = base_ring(C)(1)
+  _set_even_odd_coefficients!(res)
+  return res
+end
+
+@doc raw"""
+    gen(C::CliffordAlgebra, i::Int) -> CliffordAlgebraElem
+
+Return the $i$-th canonical multiplicative generator of the Clifford
+algebra $C$. This is just the $i$-th basis vector of the underlying
+quadratic module.
+"""
+function gen(C::CliffordAlgebra, i::Int)
+  res = CliffordAlgebraElem(C)
+  if i <= 0
+    res.coeffs[i] #Throws a BoundsError instead of a DomainError for consistency
+  end
+  res.coeffs[2^(i - 1) + 1] = base_ring(C)(1)
+  _set_even_odd_coefficients!(res)
+  return res
+end
+
+@doc raw"""
+    basis(C::CliffordAlgebra) -> Vector{CliffordAlgebraElem}
+
+Return the canonical basis of the Clifford algebra $C$.
+"""
+basis(C::CliffordAlgebra) = map(i -> basis(C, i), 1:dim(C))
+
+@doc raw"""
+    gens(C::CliffordAlgebra) -> Vector{CliffordAlgebraElem}
+
+Return the vector of canonical multiplicative generators of the Clifford algebra $C$.
```

Comment:
What are multiplicative generators? Generators of the unite group of C?

For `AbstractAssociativeAlgebra` this returns algebra generators (over the base field). If this here is doing something else, the needs to be different.

Reply:
Description clarified in this [commit](https://github.com/oscar-system/Oscar.jl/pull/4370/commits/d4b78c05016fbec7d6ab36f6b18b4b8bec2d084d)